vignettes/symptom-surveys.Rmd

```diff
@@ -0,0 +1,672 @@
+    trainer = quantile_reg(),
+    args_list = arx_args_list(
+      lags = lags,
+      ahead = .x,
+      nonneg = FALSE
+    )
+  )$predictions %>%
+    mutate(lead = .x) %>%
+    left_join(z_te %>% filter(target_date == (date + .x)), by = c("geo_value", "target_date"))) %>%
+    list_rbind() %>%
+    mutate(err3 = abs(inv_trans(.pred) - inv_trans(target_case))) %>%
+    select(geo_value, forecast_date, err3, lead)
+
+  # Cases, Facebook and Google model
+  out_df4 <- map(leads, ~ arx_forecaster(
+    z %>%
+      filter(between(time_value, date - .x - max(lags) - n, date)) %>%
+      select(time_value, geo_value, case, fb, goog) %>%
+      filter(complete.cases(.)),
+    outcome = "case",
+    predictors = c("case", "goog"),
+    trainer = quantile_reg(),
+    args_list = arx_args_list(
+      lags = lags,
+      ahead = .x,
+      nonneg = FALSE
+    )
+  )$predictions %>%
+    mutate(lead = .x) %>%
+    left_join(z_te %>% filter(target_date == (date + .x)), by = c("geo_value", "target_date"))) %>%
+    list_rbind() %>%
+    mutate(err4 = abs(inv_trans(.pred) - inv_trans(target_case))) %>%
+    select(geo_value, forecast_date, err4, lead)
+
+  # Left join of the results for all models
+  out_list[[k]] <- left_join(left_join(left_join(left_join(out_df0, out_df1), out_df2), out_df3), out_df4)
+}
+# Outside of loop bind rows of list
+out_df <- do.call(rbind, out_list)
+```
+
+## Results: All Four Models
+
+Since there are only two common forecast dates available for the four models for the 14-day-ahead forecasts (May 13 and May 14, 2020), we skip studying the 14-day-ahead forecast results in this four-way model discussion. 
+
+Below we compute the median scaled errors for each of the four models over the 9-day test period. We can see that adding either or both of the survey signals improves on the median scaled error of the model that uses cases only, with the biggest gain achieved by the "Cases + Google" model. We can also see that the median scaled errors are all close to 1 (with all but that from the "Cases + Google" and "Cases + Facebook + Google" models exceeding 1), which speaks to the difficulty of the forecasting problem.
+
+```{r}
+library(dplyr)
+library(tidyr)
+library(ggplot2)
+
+model_names <- c(
+  "Cases", "Cases + Facebook", "Cases + Google",
+  "Cases + Facebook + Google"
+)
+
+# Calculate the scaled errors for each model, that is, the error relative to the strawman's error
+res_all4 <- out_df %>%
+  drop_na() %>% # Restrict to common time
+  mutate(
+    err1 = err1 / err0, err2 = err2 / err0, # Compute relative error
+    err3 = err3 / err0, err4 = err4 / err0
+  ) %>% # to strawman model
+  mutate(
+    dif12 = err1 - err2, dif13 = err1 - err3, # Compute differences
+    dif14 = err1 - err4
+  ) %>% # relative to cases model
+  ungroup() %>%
+  select(-err0)
+
+# Calculate and print median errors, for all 4 models, and just 7 days ahead
+res_err4 <- res_all4 %>%
+  select(-starts_with("dif")) %>%
```

Comment:
This will need to change slightly in light of the above suggested simplifications